src/http/client/tls.rs

```diff
@@ -182,31 +184,43 @@ impl<'a, 'ctx> TlsConnector<'a, 'ctx> {
     /// Creates a new `TlsConnector` that will create a new `SslContext` before
     /// trying to establish the TLS connection. The path to the CA file that the
     /// context will use needs to be provided.
-    pub fn new<P: AsRef<Path>>(host: &'a str, ca_file_path: &'ctx P) -> TlsConnector<'a, 'ctx> {
+    pub fn new<P: AsRef<Path>>(host: &'a str, port:u16, ca_file_path: &'ctx P) -> TlsConnector<'a, 'ctx> {
         TlsConnector {
             host: host,
+            port: port,
             context: Http2TlsContext::CertPath(ca_file_path.as_ref()),
         }
     }
 
     /// Creates a new `TlsConnector` that will use the provided context to
     /// create the `SslStream` that will back the HTTP/2 connection.
-    pub fn with_context(host: &'a str, context: &'ctx SslContext) -> TlsConnector<'a, 'ctx> {
+    pub fn with_context(host: &'a str, port:u16, context: &'ctx SslContext) -> TlsConnector<'a, 'ctx> {
         TlsConnector {
             host: host,
+            port: port,
             context: Http2TlsContext::Wrapped(context),
         }
     }
 
     /// Builds up a default `SslContext` instance wth TLS settings that the
     /// HTTP/2 spec mandates. The path to the CA file needs to be provided.
-    pub fn build_default_context(ca_file_path: &Path) -> Result<SslContext, TlsConnectError> {
+    pub fn build_default_context(ca_file_path: &Path, verify_peer : bool, cert_file: Option<PathBuf>, private_key: Option<PathBuf>) -> Result<SslContext, TlsConnectError> {
```

Comment:
This was meant to be a _default_ context, which would be sensible without requiring the user to figure out what various switches do. That said, I'm not opposed to having a helper that offers more customization, without callers having to figure out how to build raw `SslContext`s.

Would you mind renaming this method to something like `build_context` and then implementing the original `build_default_context` in terms of that?